Cargo.toml

```diff
@@ -11,11 +11,12 @@ license = "MIT/Apache-2.0"
 readme = 'README.md'
 
 [dependencies]
+error-chain = "^0.12.0"
```

Comment:
I've used error-chain before, but many people seems to be moving to `failure` nowadays. Any reason why you prefer error-chain?

Reply:
I only used it because I was the one I knew, let me know if you want to use failure and I can change it.

Reply:
No strong opinion, replace only if you want to use it as an excuse to learn `failure` =]